MapboxNavigation/NavigationViewController.swift

```diff
@@ -389,191 +379,137 @@ open class NavigationViewController: UIViewController {
     ///   - voiceController: Produces voice instructions for route navigation. If nil, a default will be used.
     ///
     /// See [Mapbox Directions](https://mapbox.github.io/mapbox-navigation-ios/directions/) for further information.
-    @objc(initWithRoute:dayStyleURL:nightStyleURL:directions:routeController:locationManager:voiceController:)
-    public convenience init(for route: Route,
-                            dayStyleURL: URL,
+    @objc(initWithDayStyleURL:nightStyleURL:directions:routeController:locationManager:voiceController:)
+    public convenience init(dayStyleURL: URL,
                             nightStyleURL: URL? = nil,
                             directions: Directions = Directions.shared,
                             routeController: RouteController? = nil,
                             locationManager: NavigationLocationManager? = nil,
-                            voiceController: RouteVoiceController? = nil) {
+                            voiceController: RouteVoiceController = RouteVoiceController()) {
```

Comment:
Right now we offer 3 or 4 different init options. This means you can still start a navigation the old way with presenting the sheet but this includes maintenance complexity.

I would prefer to only offer one - the new way. I'm aware that this will break compatibility, however this is a big feature that warrants a major version increase, so I think we should take the opportunity to cleanup. Whats your thoughts?

Reply:
> Right now we offer 3 or 4 different init options.

Lets break this down a bit.

init 1: initWithCoder.

I literally don't know if this is usable. I think we just have it to satisfy the compiler. If it's not intended to be usable, we'd be better off replacing it with a useful `fatalError("use foo instead")`.

init 2: `@objc(initWithRoute:directions:styles:routeController:locationManager:voiceController:)` was recently deprecated (#45) because it relies on an implicit Style/tileserver (formerly this hit mapbox, now maplibre demo). It is useful to keep around because it will provide users with the upgrade path to the new init method.

init 3: Accepts a `Style`. I think we have to keep this one as it's the most general.

init 4: Most people don't care about the particulars of `Style`, and just want to give a style URL. I think it's the one most people will actually want to use.

So in summary, yes it's 4 inits, but I think we should keep them all. 

I think we should delete the deprecated method after it's been released long enough for the deprecation message to be useful. I won't be super mad though if you feel strongly about deleting it before the deprecation has been released.

I feel more strongly that we should keep both the `Style` and `styleURL` init flavors.

Disclaimer: I am biased because I deprecated init 2 and wrote init 3 + 4. 😉

Reply:
I'm a bit unhappy with the first init (deprecated) as we need to add backwards compatibility and that means double testing and special workarounds.

New way: `NavigationViewController.init(dayStyleURL: URL)`, add it as rootViewController and call `startNavigation()` when you are ready.

Old way: `NavigationViewController.init(for: route)` and this starts a navigation right away, user is expected to call `present(navigationViewController, animated: true)`. I think this currently doesn't work correctly as I've removed viewWillAppear & viewWillDisappear methods. So keeping this old init around adds a lot of complexity to keep both ways around.

Given that this will be in a major version release I think it's acceptable to break things and as long as we mention this in the release notes it shouldn't cause issues. But maybe I'm missing something important here and it does make sense to add this extra complexity.